src/Microsoft.AspNetCore.Sockets/Internal/ChannelConnection.cs

```diff
@@ -11,20 +11,23 @@ namespace Microsoft.AspNetCore.Sockets.Internal
 {
     public class ChannelConnection<T> : IChannelConnection<T>
     {
-        public IReadableChannel<T> Input { get; }
-        public IWritableChannel<T> Output { get; }
+        public IChannel<T> Input { get; }
```

Comment:
It can be private, it just needs to be typed as `IChannel<T>` in this class so we can complete it.